app/gui/src/project-view/components/shared/AgGridTableView/Utils.ts

```diff
@@ -120,3 +120,61 @@ export const getAgGridProperties = (): [Properties, Properties, Properties] => {
 
   return [props, computed, watch]
 }
+
+export function createDateTime(x: {
+  year?: number
+  month?: number
+  day?: number
+  hour?: number
+  minute?: number
+  second?: number
+  type?: string
+}): Date {
+  return new Date(Date.UTC(
+    x.year ?? 1970,
+    (x.month ?? 1) - 1,
+    x.day ?? 1,
+    x.hour ?? 0,
+    x.minute ?? 0,
+    x.second ?? 0
+  ))
+}
+
+export function formatDateLikeValue(value: any): string {
+  const valueType = value?.type
+  const date = createDateTime(value)
+  if (valueType === 'Date') {
+    return date.toISOString().split('T')[0]
+  } else if (valueType === 'Date_Time') {
+    return date.toISOString().replace('T', ' ').replace('Z', '')
+  } else if (valueType === 'Time_Of_Day') {
+    const timeString = date.toISOString().split('T')[1]
+    return timeString ? timeString.replace('Z', '') : ''
+  }
+  return ''
+}
+
+export function ensoDateComparator(a: any, b: any): number {
```

Comment:
If we deserialised the object into JS Date, would we need this?

Reply:
If we stored them as JS Date's behind AGGrid then I assume not. But then we would also be unable to distinguish between the 3 time based types.